kotlinpoet/src/commonMain/kotlin/com/squareup/kotlinpoet/NameAllocator.kt

```diff
@@ -77,7 +77,39 @@ public class NameAllocator private constructor(
   private val allocatedNames: MutableSet<String>,
   private val tagToName: MutableMap<Any, String>,
 ) {
-  public constructor() : this(mutableSetOf(), mutableMapOf())
+  public constructor() : this(preAllocateKeywords = true)
+
+  /**
+   * @param preAllocateKeywords If true, all Kotlin keywords will be pre-allocated and suffixed with
+   * underscores to avoid using them as identifiers:
+   *
+   * ```kotlin
+   * val nameAllocator = NameAllocator(preAllocateKeywords = true)
+   * println(nameAllocator.newName("when")) // prints "when_"
+   * ```
+   *
+   * If false, keywords will not get any special treatment:
+   *
+   * ```kotlin
+   * val nameAllocator = NameAllocator(preAllocateKeywords = false)
+   * println(nameAllocator.newName("when")) // prints "when"
+   * ```
+   *
+   * Note that you can use the `%N` placeholder when emitting a name produced by [NameAllocator] to
+   * ensure it's properly escaped for use as an identifier:
+   *
+   * ```kotlin
+   * val nameAllocator = NameAllocator(preAllocateKeywords = false)
+   * println(CodeBlock.of("%N", nameAllocator.newName("when"))) // prints "`when`"
+   * ```
+   *
+   * The default behaviour of [NameAllocator] is to pre-allocate keywords - this is the behaviour you'll
+   * get when using the no-arg constructor.
+   */
+  public constructor(preAllocateKeywords: Boolean) : this(
```

Comment:
I think "pre-allocate" means "before allocate" and "preallocate" means "allocate in advance" so we want the casing here and elsewhere in the change to be `preallocateKeywords`.

Reply:
Ah, didn't realize "preallocate" can be a single word, thanks! [2d690ae](https://github.com/square/kotlinpoet/pull/1803/commits/2d690ae28bd662948487c69d54fa1959f5aa9a01)

Reply:
English is severely underspecified! I'm still not 100% sure of this, but most signs point toward "preallocate" being the more correct form. It was a weird rabbit hole to fall into this morning while walking the dog...

Reply:
Lol. English needs official documentation.